docs/doc_sources/api_reference/dpctl/index.rst

```diff
@@ -0,0 +1,86 @@
+.. _dpctl_pyapi:
+
+.. currentmodule:: dpctl
+
+:py:mod:`dpctl`
+===============
+
+.. py:module:: dpctl
+
+.. rubric:: Submodules
+
+.. list-table::
+    :widths: 10 50
+
+    * - :py:mod:`dpctl.memory`
+      - Unified Shared Memory operations
+    * - :py:mod:`dpctl.program`
+      - Support for working with SYCL kernels
+    * - :py:mod:`dpctl.tensor`
+      - Array library conforming to Python Array API specification
+    * - :py:mod:`dpctl.utils`
+      - A collection of utility functions
+
+.. rubric:: Classes
+
+.. autosummary::
+    :toctree: generated
+    :nosignatures:
+    :template: autosummary/cython_class.rst
+
+    SyclDevice
+    SyclContext
+    SyclQueue
+    SyclEvent
+    SyclPlatform
+    SyclTimer
+
+.. rubric:: Device selection
+
+.. _dpctl_device_selection_functions:
+
+.. autosummary::
+    :toctree: generated
+    :nosignatures:
+
+    select_default_device
+    select_cpu_device
+    select_gpu_device
+    select_accelerator_device
+    select_device_with_aspects
```

Comment:
This function could use a proper docstring later.

Reply:
This was referring specifically to `select_device_with_aspects`, which has a description that isn't a true docstring.